hal_core/src/mm.rs

```diff
@@ -136,41 +164,47 @@ pub fn align_up(val: usize, page_sz: usize) -> usize {
     ((val + page_sz - 1) / page_sz) * page_sz
 }
 
-pub fn init_paging<P: PageMap + 'static>(
+pub fn align_down(addr: usize, page_sz: usize) -> usize {
+    // TODO: can this be more optimized ?
+    // XXX: uh isn't this math wrong ?
+    align_up(addr, page_sz) + page_sz
```

Comment:
I think the math is wrong indeed
```suggestion
    align_up(addr, page_sz) - page_sz
```

Reply:
I know how to fix the math.
The issue is that if we fix it everything breaks ^^